Core/MIPS/IR/IRJit.cpp

```diff
@@ -0,0 +1,430 @@
+// Copyright (c) 2012- PPSSPP Project.
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, version 2.0 or later versions.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License 2.0 for more details.
+
+// A copy of the GPL 2.0 should have been included with the program.
+// If not, see http://www.gnu.org/licenses/
+
+// Official git repository and contact information can be found at
+// https://github.com/hrydgard/ppsspp and http://www.ppsspp.org/.
+
+#include "base/logging.h"
+#include "profiler/profiler.h"
+#include "Common/ChunkFile.h"
+#include "Common/CPUDetect.h"
+#include "Common/StringUtils.h"
+
+#include "Core/Reporting.h"
+#include "Core/Config.h"
+#include "Core/Core.h"
+#include "Core/CoreTiming.h"
+#include "Core/Debugger/SymbolMap.h"
+#include "Core/MemMap.h"
+
+#include "Core/MIPS/MIPS.h"
+#include "Core/MIPS/MIPSCodeUtils.h"
+#include "Core/MIPS/MIPSInt.h"
+#include "Core/MIPS/MIPSTables.h"
+#include "Core/HLE/ReplaceTables.h"
+#include "Core/HLE/sceKernelMemory.h"
+#include "Core/MIPS/IR/IRRegCache.h"
+#include "Core/MIPS/IR/IRJit.h"
+#include "Core/MIPS/IR/IRPassSimplify.h"
+#include "Core/MIPS/JitCommon/JitCommon.h"
+
+namespace MIPSComp
+{
+
+IRJit::IRJit(MIPSState *mips) : mips_(mips) { 
+	logBlocks = 0;
+	dontLogBlocks = 0;
+	js.startDefaultPrefix = mips_->HasDefaultPrefix();
+	js.currentRoundingFunc = convertS0ToSCRATCH1[0];
+	u32 size = 128 * 1024;
+	blTrampolines_ = kernelMemory.Alloc(size, true, "trampoline");
```

Comment:
Hmm, I think this isn't used, I see where you're going here.  Would take careful invalidation handling...

Maybe we should comment out / TODO as long as this memory isn't being used?

-[Unknown]